core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -195,15 +198,54 @@ public Result visit(Project e) {
     final Builder builder =
         x.builder(e, Clause.SELECT);
     final List<SqlNode> selectList = new ArrayList<>();
+
+    boolean checkNull = isCastNullNeeded(stack);
```

Comment:
to be consistent, please declare `checkNull` variable `final`

Reply:
I wil fix it.

Reply:
I have fixed it and commit has been pushed.